### ISSUE TITLE
Improvements for `JmsEndpointConfigParser` and `JmsSyncEndpointConfigParser`

### ISSUE THREAD

**Issue by picimako** [citrusframework/citrus#1066]
Hi!

I'm looking into how `JmsEndpointConfigParser` and `JmsSyncEndpointConfigParser` works, so that I could implement some IntelliJ inspections for their respective config annotations, and I noticed a few things that could be improved in and regarding them:

- I cannot see a way in which the exception *"Required connection-factory is missing for jms configuration"* would be thrown. Since `connectionFactory` has a default value, `if (!StringUtils.hasText(connectionFactory))` will never be true.
```java
//connectionFactory
String connectionFactory = "connectionFactory"; //default value

if (StringUtils.hasText(annotation.connectionFactory())) {
    connectionFactory = annotation.connectionFactory();
}

if (!StringUtils.hasText(connectionFactory)) {
    throw new CitrusRuntimeException("Required connection-factory is missing for jms configuration");
}
```

- Regarding the exception *""When providing a jms-template, none of connection-factory, destination, or destination-name should be provided"*:
  - This is thrown in the else branch of `StringUtils.hasText(destination) || StringUtils.hasText(destinationName)`, so in that else branch both `destination` and `destinationName` are empty, thus checking for `StringUtils.hasText(destination) || StringUtils.hasText(destinationName)` there seems unnecessary.
  - I think the destination and destination name checks should be removed from the else branch, and the `jmsTemplate` text check/branch should be copied to the `if (StringUtils.hasText(destination) || StringUtils.hasText(destinationName))` block of the condition, **something** like this:

```java
if (StringUtils.hasText(destination) || StringUtils.hasText(destinationName)) {
    //...

    if (StringUtils.hasText(jmsTemplate)) {
        //...
    }
} else if (StringUtils.hasText(jmsTemplate)) {
    if (StringUtils.hasText(annotation.connectionFactory())) {
        throw new CitrusRuntimeException("When providing a jms-template, none of " +
                "connection-factory, destination, or destination-name should be provided");
    }

    builder.jmsTemplate(referenceResolver.resolve(jmsTemplate, JmsTemplate.class));
}
```

- The exception branches of the conditions are not covered by unit tests, so those could be added.

Let me know what you think.

**Comment by bbortt** [comment]
> I cannot see a way in which the exception "Required connection-factory is missing for jms configuration" would be thrown. Since connectionFactory has a default value, if (!StringUtils.hasText(connectionFactory)) will never be true.

that is true.

> This is thrown in the else branch of StringUtils.hasText(destination) || StringUtils.hasText(destinationName), so in that else branch both destination and destinationName are empty, thus checking for StringUtils.hasText(destination) || StringUtils.hasText(destinationName) there seems unnecessary.

also true.

> The exception branches of the conditions are not covered by unit tests, so those could be added.

**must** be added 😉 coverage is generally too low for sure.

**Comment by picimako** [comment]
Alright. I'll probably open a PR in the next few days with the changes and some extra tests too. :)

**Comment by christophd** [comment]
Fixed with #1068
